Source/Widgets/Viewer/Viewer.js

```diff
@@ -1855,19 +1872,56 @@ Either specify options.terrainProvider instead or set options.baseLayerPicker to
     };
 
     function updateZoomTarget(viewer) {
-        var entities = viewer._zoomTarget;
-        if (!defined(entities) || viewer.scene.mode === SceneMode.MORPHING) {
+        var target = viewer._zoomTarget;
+        if (!defined(target) || viewer.scene.mode === SceneMode.MORPHING) {
             return;
         }
 
         var scene = viewer.scene;
         var camera = scene.camera;
         var zoomPromise = viewer._zoomPromise;
         var zoomOptions = defaultValue(viewer._zoomOptions, {});
+        var options;
+
+        // If zoomTarget was Cesium3DTileset
+        if (defined(target.readyPromise)) {
+            return target.readyPromise.then(function() {
+                var bSphere = target.boundingSphere;
+                // if offset was originally undefined then give it base value instead of empty object
+                if (!defined(zoomOptions.offset)) {
+                    zoomOptions.offset = new HeadingPitchRange(0.0, 0.0, 2.0 * bSphere.radius);
+                }
+
+                options = {
```

Comment:
Make sure you are taking into account the following options that can be passed to the flyTo function.

```
      * @param {Number} [options.duration=3.0] The duration of the flight in seconds.
      * @param {Number} [options.maximumHeight] The maximum height at the peak of the flight.
```

Add a tests for when these are provided, they are being passed to `flyToBoundingSphere`.